pandas/core/indexes/base.py

```diff
@@ -404,6 +404,11 @@ def __new__(
                     )
             # other iterable of some kind
             subarr = com.asarray_tuplesafe(data, dtype=object)
+
+            if isinstance(name, list) and len(name) == 1:
```

Comment:
we do not want to add this here, this is a very complicated routine; what are you trying to accomplish?

Reply:
I was looking for the functionality of `Index._get_names()` / `MultiIndex._get_names()`

Reverted this + using `_get_names` now